MobileTraining/MobileTraining/Networking/NetworkManager.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  NetworkManager.swift
+//  MobileTraining
+//
+//  Created by Svetoslav Donkov on 16.05.23.
+//
+
+import Foundation
+import Alamofire
+import KeychainAccess
+
+class NetworkManager {
+    private let session: Session
+    let interceptor = Interceptor()
+    
+    init() {
+        // Create custom session configuration
+        let configuration = URLSessionConfiguration.default
+        configuration.timeoutIntervalForRequest = 30 // Set custom timeout
+        
+        // Initialize the custom session with the configuration
+        self.session = Session(configuration: configuration, interceptor: interceptor)
+    }
+    
+    func request<T: Decodable>(_ url: String, method: HTTPMethod = .get, parameters: [String: Any]? = nil) async throws -> T {
+        try await withCheckedThrowingContinuation { continuation in
+            session.request(url, method: method, parameters: parameters)
+                .validate()
+                .responseDecodable(of: T.self) { response in
+                    switch response.result {
+                    case .success(let data):
+                        continuation.resume(returning: data)
+                    case .failure(let error):
+                        continuation.resume(throwing: error)
+                    }
+                }
+        }
+    }
+}
+
+struct Interceptor: RequestInterceptor {
+    private let keychain = Keychain(service: "com.accedia.svdo.MobileTraining.keychain")
```

Comment:
Nice that you've thought of using the keychain but ideally it would come from a separate Factory injected service that you build, specifically for keychain storage. Placing the keychain storing responsibility in the interceptor itself isn't very clean.